packages/actor-query-operation-filter-tree/lib/SparqlExpressionEvaluator.ts

```diff
@@ -0,0 +1,241 @@
+import {Bindings} from "@comunica/bus-query-operation";
+import * as RDF from "rdf-js";
+import {stringToTerm, termToString} from "rdf-string";
+import {Algebra} from "sparqlalgebrajs";
+
+export class SparqlExpressionEvaluator {
```

Comment:
This file is identical to the evaluator in the other filter actor, which is a bad thing. There also seems to be some overlap between the 2 main actor classes. If both classes share so much there needs to be some sort of inheritance somewhere to prevent this duplication (possibly with an abstract class).

Even better might be, since this class uses so much of the other actor, to actually use both actors. I.e., have this actor do some changes to the query and context, and then call the other filter actor through the mediator with that new query. This would remove all duplication that is currently required.